src/pybel/schema/__init__.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+
+"""Validation for PyBEL data."""
+
+import json
+import os
+from typing import Any, Mapping, Optional, Tuple
+
+import jsonschema
+
+__all__ = ['is_valid_node']
+
+HERE = os.path.abspath(os.path.dirname(__file__))
+
+
+def _load_schema(filename: str) -> Mapping[str, Any]:
+    """
+    Load a schema from its filename.
+
+    :param filename: The relative path to the schema, e.g. "variant.schema.json"
+    """
+    path = os.path.join(HERE, filename)
+    with open(path) as json_schema:
+        schema = json.load(json_schema)
+    return schema
+
+
+def _validator(filename: str) -> jsonschema.Draft7Validator:
+    """
+    Return a validator that checks against a given schema.
+
+    :param filename: The relative path to the schema, e.g. "variant.schema.json"
+    """
+    schema = _load_schema(filename)
+    # To use schemas from other files, jsonschema needs to know where the references point to, so
+    # create a resolver that directs any references (like "fusion.schema.json") to the schema's dir
+    schema_uri = 'file://' + HERE + '/'
+    resolver = jsonschema.RefResolver(base_uri=schema_uri, referrer=__file__)
+    return jsonschema.Draft7Validator(schema, resolver=resolver)
+
+
+def _get_schema(node: Mapping[str, Any]) -> Tuple[str, bool]:
+    """
+    Get the filename for the appropriate schema for a given node.
+
+    :return: The filename for the schema and an error bool.
+    """
+    err = False
+    filename = ""
+    # Check if the node is a subclass of FusionBase
+    if 'fusion' in node.keys():
+        filename = 'fusion.schema.json'
+    # Check if the "function" is present (if not, the node is invalid)
+    elif 'function' in node.keys():
+        if node['function'] == 'Reaction':
+            filename = 'reaction.schema.json'
+        elif node['function'] in ['Complex', 'Composite']:
+            filename = 'list_abundance.schema.json'
+        else:
+            filename = 'base_abundance.schema.json'
+    else:
+        err = True
```

Comment:
should raise an error since there's no sensical way to get here

Reply:
Since the conditional logic is now in the JSON file, this is no longer an issue